Boost: Use My Jetpack Pricing Table View

nateweller opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

My Jetpack now includes the pricing table for Boost. This PR updates Boost to direct the "getting started" flow to the product's pricing table screen in My Jetpack.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Update Boost product data in the My Jetpack package.
* Remove `GettingStarted` component from Boost.
* Replace the redirect to the `/getting-started` page to `/admin.php?page=my-jetpack#/add-boost`.
* Bumps project versions from `tools/fixup-project-versions.sh`.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

https://github.com/Automattic/jetpack/pull/29931

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

The pricing table in Boost does include custom events such as `free_cta_from_getting_started_page_in_plugin`, which would be removed by this PR.

The equivalent action would now be tracked in My Jetpack via `jetpack_myjetpack_product_interstitial_add_link_click`.

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Start with a fresh site with only Jetpack Boost installed.
* Visit the Jetpack Boost admin page. Validate that you are redirected to the pricing table in My Jetpack.
* Activate Boost by choosing one of the plans. Validate that you are returned to the Boost admin screen.

nateweller committed:
[not verified] Use the pricing interstitial from My Jetpack in the Boost plugin.

nateweller committed:
[not verified] Fixup project versions

thingalon reviewed (changes requested):
I've noticed a couple of issues:

When I select the free plan, I get stuck in a UI loop:
https://github.com/Automattic/jetpack/assets/1369626/e8e1a54c-1f1e-491d-ba02-936a55b93e27

Some errors appear in the console when you click on the (i) icon next to Image CDN. I think there is some DOM sadness:
![image](https://github.com/Automattic/jetpack/assets/1369626/837d3af9-8960-4f68-a0ee-f2c074ee8574)

Otherwise, I think this looks good

nateweller committed:
[not verified] Ensure the site is connected during the activation process in ProductDetailTable

nateweller committed:
[not verified] Update PricingTable to use a more generic wrapper for tooltip content

nateweller committed:
[not verified] changelog

nateweller commented:
Thanks for your review @thingalon!

The UI loop was due to the "Start for Free" button not taking any action beyond redirecting the user to the plugin screen. I've added logic so that is creates a site-level connection when starting for free.

I've also updated the Pricing Table component to allow more complex tooltip info content, by swapping a `p` for a `div` to prevent DOM nesting errors :+1:

thingalon commented:
Thanks for the updates, @nateweller. I can confirm that it seems to be working much better now.

It looks like the Boost e2e tests are currently broken - once those are happy, I think we can go ahead with this :)

nateweller committed:
[not verified] Update e2e test
